### ISSUE TITLE
Avogadro2 crash

### ISSUE THREAD

**Issue by kevinsmia1939** [flathub/org.openchemistry.Avogadro2#47]
Avogadro2 crash
testing with eigen 339
Eigen 3.3.9 and 3.4.0 failed.
```
andy@localhost:~> flatpak run org.openchemistry.Avogadro2
QIBusPlatformInputContext: invalid portal bus.
Locale:  "en_US"
Extension plugins dynamically found... 37
Checking for  "commands"  scripts in "/home/andy/.var/app/org.openchemistry.Avogadro2/data/OpenChemistry/Avogadro/commands"
Checking for  "commands"  scripts in "/app/share/OpenChemistry/Avogadro/commands"
Checking for  "commands"  scripts in "/usr/share/OpenChemistry/Avogadro/commands"
Checking for  "commands"  scripts in "/usr/share/runtime/share/OpenChemistry/Avogadro/commands"
Checking for  "commands"  scripts in "/run/host/share/OpenChemistry/Avogadro/commands"
Checking for  "commands"  scripts in "/app/bin/../lib/avogadro2/scripts/commands"
OBProcess::executeObabel: Running "/app/bin/obabel" "-L formats read"
OBProcess::executeObabel: Running "/app/bin/obabel" "-L formats write"
OBProcess::executeObabel: Running "/app/bin/obabel" "-L forcefields"
OBProcess::executeObabel: Running "/app/bin/obabel" "-L charges"
OBProcess::executeObabel: Running "/app/bin/obabel" "-V"
"/app/bin/obabel"  found:  "/app/bin/obabel: Open Babel 3.1.0 -- Dec 31 2021 -- 22:11:37"
Checking for  "inputGenerators"  scripts in "/home/andy/.var/app/org.openchemistry.Avogadro2/data/OpenChemistry/Avogadro/inputGenerators"
Checking for  "inputGenerators"  scripts in "/app/share/OpenChemistry/Avogadro/inputGenerators"
Checking for  "inputGenerators"  scripts in "/usr/share/OpenChemistry/Avogadro/inputGenerators"
Checking for  "inputGenerators"  scripts in "/usr/share/runtime/share/OpenChemistry/Avogadro/inputGenerators"
Checking for  "inputGenerators"  scripts in "/run/host/share/OpenChemistry/Avogadro/inputGenerators"
Checking for  "inputGenerators"  scripts in "/app/bin/../lib/avogadro2/scripts/inputGenerators"
Checking for  "formatScripts"  scripts in "/home/andy/.var/app/org.openchemistry.Avogadro2/data/OpenChemistry/Avogadro/formatScripts"
Checking for  "formatScripts"  scripts in "/app/share/OpenChemistry/Avogadro/formatScripts"
Checking for  "formatScripts"  scripts in "/usr/share/OpenChemistry/Avogadro/formatScripts"
Checking for  "formatScripts"  scripts in "/usr/share/runtime/share/OpenChemistry/Avogadro/formatScripts"
Checking for  "formatScripts"  scripts in "/run/host/share/OpenChemistry/Avogadro/formatScripts"
Checking for  "formatScripts"  scripts in "/app/bin/../lib/avogadro2/scripts/formatScripts"
"Navigator" added
"Editor" added
"Selection" added
"Manipulator" added
"BondCentric" added
"MeasureTool" added
"PlayerTool" added
Open Babel formats ready:  142
/usr/include/c++/10.2.0/bits/stl_vector.h:1045: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](std::vector<_Tp, _Alloc>::size_type) [with _Tp = Eigen::Matrix<double, 3, 1>; _Alloc = std::allocator<Eigen::Matrix<double, 3, 1> >; std::vector<_Tp, _Alloc>::reference = Eigen::Matrix<double, 3, 1>&; std::vector<_Tp, _Alloc>::size_type = long unsigned int]: Assertion '__builtin_expect(__n < this->size(), true)' failed.
```

**Comment by ghutchis** [comment]
I'm not sure the cause of this crash, but we fixed a pile of them in the 1.96 release.

**Comment by kevinsmia1939** [comment]
> I'm not sure the cause of this crash, but we fixed a pile of them in the 1.96 release.

I test the new version, and it does not crash anymore now.
